paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -62,6 +64,22 @@ class PaparazziPlugin : Plugin<Project> {
         project.tasks.named("compile${testVariantSlug}Kotlin")
             .configure { it.dependsOn(writeResourcesTask) }
       }
+
+      val recordTaskProvider = project.tasks.register("recordPaparazzi${variantSlug}") {
+        System.setProperty("paparazzi.record", "true")
+      }
+
+      val verifyTaskProvider = project.tasks.register("verifyPaparazzi${variantSlug}") {
+        System.setProperty("paparazzi.verify", "true")
+      }
+
+      val testTaskProvider = project.tasks.named("test${testVariantSlug}", Test::class.java) {
+        it.systemProperty("paparazzi.test.record", System.getProperty("paparazzi.record") ?: "false")
+        it.systemProperty("paparazzi.test.verify", System.getProperty("paparazzi.verify") ?: "false")
```

Comment:
System.setProperty lets the gradle task invoker set the property, but tests run in a separate forked JVM, so this pipes that setting through.

Reply:
What about using `gradle.taskGraph.hasTask("record${slug}")` instead of using task configuration to side-effect setting a system property for this to be enabled.

Reply:
Two thoughts on that:
1) I think peeking into the taskGraph is considered a smell by the Gradle team.
2) This would couple test run configuration to Gradle.  While it's our preferred build tool, it would be nice to enable extension/contributions from others for non-Gradle build systems.